test/common/network/dns_impl_test.cc

```diff
@@ -107,31 +125,81 @@ class TestDnsServerQuery {
+        // and lookup the ips of the cname and send back those ips (if any) too
+        auto cit = parent_.cnames_.find(name);
+        if (cit != parent_.cnames_.end()) {
+          cname = cit->second;
+        }
+        const char* hostLookup = name;
+        const unsigned char* ip_question = question;
+        long ip_name_len = name_len;
+        std::string encodedCname;
+        if (cname.size() > 0) {
+          hostLookup = cname.c_str();
+          encodedCname = TestDnsServerQuery::encodeDnsName(cname);
+          ip_question = reinterpret_cast<const unsigned char*>(encodedCname.c_str());
+          ip_name_len =
+              encodedCname.size() + 1; //+1 as we need to include the final null terminator
+        }
         ASSERT_TRUE(q_type == T_A || q_type == T_AAAA);
         if (q_type == T_A) {
-          auto it = parent_.hosts_A_.find(name);
+          auto it = parent_.hosts_A_.find(hostLookup);
           if (it != parent_.hosts_A_.end()) {
             ips = &it->second;
           }
         } else {
-          auto it = parent_.hosts_AAAA_.find(name);
+          auto it = parent_.hosts_AAAA_.find(hostLookup);
           if (it != parent_.hosts_AAAA_.end()) {
             ips = &it->second;
           }
         }
         ares_free_string(name);
 
+        int answer_size = ips != nullptr ? ips->size() : 0;
+        answer_size += encodedCname.size() > 0 ? 1 : 0;
+
         // The response begins with the intial part of the request
         // (including the question section).
         const size_t response_base_len = HFIXEDSZ + name_len + QFIXEDSZ;
         unsigned char response_base[response_base_len];
         memcpy(response_base, request, response_base_len);
         DNS_HEADER_SET_QR(response_base, 1);
         DNS_HEADER_SET_AA(response_base, 0);
-        DNS_HEADER_SET_RCODE(response_base, ips != nullptr ? NOERROR : NXDOMAIN);
-        DNS_HEADER_SET_ANCOUNT(response_base, ips != nullptr ? ips->size() : 0);
+        DNS_HEADER_SET_RCODE(response_base, answer_size > 0 ? NOERROR : NXDOMAIN);
+        DNS_HEADER_SET_ANCOUNT(response_base, answer_size);
         DNS_HEADER_SET_NSCOUNT(response_base, 0);
         DNS_HEADER_SET_ARCOUNT(response_base, 0);
+        // Total response size will be computed according to cname response size + ip response sizes
+        size_t response_ip_rest_len;
+        if (q_type == T_A) {
+          response_ip_rest_len =
+              ips != nullptr ? ips->size() * (ip_name_len + RRFIXEDSZ + sizeof(in_addr)) : 0;
+        } else {
+          response_ip_rest_len =
+              ips != nullptr ? ips->size() * (ip_name_len + RRFIXEDSZ + sizeof(in6_addr)) : 0;
+        }
+        size_t response_cname_len =
+            encodedCname.size() > 0 ? name_len + RRFIXEDSZ + encodedCname.size() + 1 : 0;
+        const uint16_t response_size_n =
+            htons(response_base_len + response_ip_rest_len + response_cname_len);
+        Buffer::OwnedImpl write_buffer_;
```

Comment:
If this is local scope you don't need the trailing _ - that's only for member variables

Reply:
eyes of a falcon, totally missed it

Reply:
The trailing "_" was present before my changes, so I didn't really give this thought. Anyway, I'll add a commit to change the name of the variable.